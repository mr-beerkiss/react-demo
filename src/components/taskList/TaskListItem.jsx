import React, {Component, PropTypes} from 'react'
import classNames from 'classnames'
import styles from './TaskListItem.css'
import deleteIcon from '../../assets/deleteIcon.svg'

export default class TaskListItem extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    complete: PropTypes.bool.isRequired,
    archived: PropTypes.bool.isRequired,
    onChange: PropTypes.func,
    onComplete: PropTypes.func,
    onArchive: PropTypes.func
  }

  static defaultProps = {
    onChange: () => {},
    onComplete: () => {},
    onArchive: () => {}
  }

  render () {
    const {
      name,
      complete,
      archived,
      onChange,
      onComplete,
      onArchive
    } = this.props

    return (
      <div className={classNames(styles.wrapper, {
        [styles.itemComplete]: complete,
        [styles.itemArchived]: archived
      })}>
        <input
          className={styles.taskText}
          type='text'
          value={name}
          disabled={complete}
          onChange={(e) => { onChange(e.target.value) }} />
        <input
          className={styles.checkbox}
          type='checkbox'
          onClick={() => { onComplete(!complete) }}
          checked={complete} />
        <img
          className={styles.deleteIcon}
          src={deleteIcon}
          alt='delete task'
          onClick={() => { onArchive(!archived) }} />
      </div>
    )
  }
}
