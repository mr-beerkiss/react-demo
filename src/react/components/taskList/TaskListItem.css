.wrapper {
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 10px 0;
}

.taskText {
  font-size: inherit;
  flex-grow: 1;
}

.deleteIcon {
  width: 20px;
  height: 20px;
  cursor: pointer;
}

.checkbox {
  cursor: pointer;
}

.itemComplete .taskText {
  text-decoration: line-through;
}
