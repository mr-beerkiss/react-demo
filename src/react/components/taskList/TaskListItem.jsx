import React, {Component, PropTypes} from 'react'
import classNames from 'classnames'
import styles from './TaskListItem.css'
import deleteIcon from '../../../assets/deleteIcon.svg'

export default class TaskListItem extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    complete: PropTypes.bool.isRequired,
    removed: PropTypes.bool.isRequired,
    onChange: PropTypes.func,
    onComplete: PropTypes.func,
    onRemove: PropTypes.func
  }

  static defaultProps = {
    onChange: () => {},
    onComplete: () => {},
    onRemove: () => {}
  }

  render () {
    const {
      name,
      complete,
      removed,
      onChange,
      onComplete,
      onRemove
    } = this.props

    return (
      <div className={classNames(styles.wrapper, {
        [styles.itemComplete]: complete,
        [styles.itemRemoved]: removed
      })}>
        <input
          className={styles.taskText}
          type='text'
          value={name}
          disabled={complete}
          onChange={(e) => { onChange(e.target.value) }} />
        <input
          className={styles.checkbox}
          type='checkbox'
          onClick={() => { onComplete(!complete) }}
          checked={complete} />
        <img
          className={styles.deleteIcon}
          src={deleteIcon}
          alt='delete task'
          onClick={() => { onRemove() }} />
      </div>
    )
  }
}
