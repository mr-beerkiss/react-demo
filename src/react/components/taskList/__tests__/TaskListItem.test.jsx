/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'
import TaskListItem from '../TaskListItem.jsx'
import styles from '../TaskListItem.css'

describe('<TaskListItem />', () => {
  it('snapshot: renders incomplete task', () => {
    const wrapper = shallow(
      <TaskListItem
        name='Task 1'
        complete={false}
        removed={false} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot: renders complete task', () => {
    const wrapper = shallow(
      <TaskListItem
        name='Task 1'
        complete
        removed={false} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot: renders deleted task', () => {
    const wrapper = shallow(
      <TaskListItem
        name='Task 1'
        complete
        removed />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('text onChange returns new task name value', () => {
    const mockFn = jest.fn()
    const wrapper = shallow(
      <TaskListItem
        name='Task 1'
        complete={false}
        removed={false}
        onChange={mockFn} />
    )

    const input = wrapper.find(`.${styles.taskText}`)
    input.simulate('change', {target: {value: 'x'}})
    expect(mockFn.mock.calls[0]).toEqual(['x'])
  })

  it('checkbox onClick returns complete state', () => {
    const mockFn = jest.fn()
    const wrapper = shallow(
      <TaskListItem
        name='Task 1'
        complete={false}
        removed={false}
        onComplete={mockFn} />
    )

    const input = wrapper.find(`.${styles.checkbox}`)
    input.simulate('click')
    expect(mockFn.mock.calls[0]).toEqual([true])
  })

  it('delete onClick triggers callback', () => {
    const mockFn = jest.fn()
    const wrapper = shallow(
      <TaskListItem
        name='Task 1'
        complete={false}
        removed
        onRemove={mockFn} />
    )

    const input = wrapper.find(`.${styles.deleteIcon}`)
    input.simulate('click')
    expect(mockFn.mock.calls[0]).toEqual([])
  })
})
