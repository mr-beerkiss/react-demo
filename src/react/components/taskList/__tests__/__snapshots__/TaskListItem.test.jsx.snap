exports[`<TaskListItem /> snapshot: renders complete task 1`] = `
<div
  className="wrapper itemComplete">
  <input
    className="taskText"
    disabled={true}
    onChange={[Function]}
    type="text"
    value="Task 1" />
  <input
    checked={true}
    className="checkbox"
    onClick={[Function]}
    type="checkbox" />
  <img
    alt="delete task"
    className="deleteIcon"
    onClick={[Function]}
    src="deleteIcon.svg" />
</div>
`;

exports[`<TaskListItem /> snapshot: renders deleted task 1`] = `
<div
  className="wrapper itemComplete">
  <input
    className="taskText"
    disabled={true}
    onChange={[Function]}
    type="text"
    value="Task 1" />
  <input
    checked={true}
    className="checkbox"
    onClick={[Function]}
    type="checkbox" />
  <img
    alt="delete task"
    className="deleteIcon"
    onClick={[Function]}
    src="deleteIcon.svg" />
</div>
`;

exports[`<TaskListItem /> snapshot: renders incomplete task 1`] = `
<div
  className="wrapper">
  <input
    className="taskText"
    disabled={false}
    onChange={[Function]}
    type="text"
    value="Task 1" />
  <input
    checked={false}
    className="checkbox"
    onClick={[Function]}
    type="checkbox" />
  <img
    alt="delete task"
    className="deleteIcon"
    onClick={[Function]}
    src="deleteIcon.svg" />
</div>
`;
